File: cfgs/pythoncfgs/FRIENDS_PAIRING_PROBLEM.py
Nodes
1;statement;foo_1 = [0 for foo_2 in fun_1(foo_3 + 1)];8;
2;conditional;fun_1(foo_1 + 1);9;
3;statement;return foo_1[foo_2];14;
4;conditional;foo_1 <= 2;10;
5;statement;foo_1[foo_2] = foo_3[foo_4 - 1] + (foo_5 - 1) * foo_6[foo_7 - 2];13;
6;statement;foo_1[foo_2] = foo_3;11;
7;exit;;
Edges
0;2;
1;3;4;
2;7;
3;5;6;
4;2;
5;2;
6;

File: cfgs/java_cfgs_foo/FRIENDS_PAIRING_PROBLEM.java
Nodes
1;special;;;
2;statement;int[] dp = new int[foo_1 + 1];14;
3;statement;int i = 0;15;
4;conditional;foo_1 <= foo_2;16;
5;conditional;foo_1 <= 2;18;
6;statement;foo_1[foo_2] = foo_3;18;
7;statement;foo_1[foo_2] = foo_3[foo_4 - 1] + ((foo_5 - 1) * foo_6[foo_7 - 2]);19;
8;statement;i++;17;
9;statement;return foo_1[foo_2];21;
10;exit;;;
Edges
0;2;
1;3;
2;4;
3;5;9;
4;6;7;
5;8;
6;8;
7;4;
8;10;
9;